joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/RubySrc2Cpg.scala

```diff
@@ -64,6 +71,10 @@ class RubySrc2Cpg extends X2CpgFrontend[Config] {
 
 object RubySrc2Cpg {
 
-  def postProcessingPasses(cpg: Cpg, config: Option[Config] = None): List[CpgPassBase] = List(new NaiveCallLinker(cpg))
+  def postProcessingPasses(cpg: Cpg, config: Option[Config] = None): List[CpgPassBase] =
+    List(
+      // new RubyTypeRecoveryPass(cpg),
```

Comment:
I recommend implementing a `RubyTypeHintCallLinker` too in this PR as it's the only way calls from types recovered in this pass really get call edges linked. It is fortunately much simpler to implement and barely requires overriding anything.

Reply:
ok, will update PR with that